integration-tests/validation.py

```diff
@@ -501,27 +501,34 @@ def validate_status_messages(model):
 
 @log_calls_async
 async def validate_docker_logins(model):
-    kubectl_cmd = '/snap/bin/kubectl --kubeconfig /root/cdk/kubeconfig'
+    # Choose a worker. He shall be our vessel.
+    app = model.applications['kubernetes-worker']
+    vessel = app.units[0]
 
-    @log_calls_async
-    async def check_output(unit, cmd):
-        action = await unit.run(cmd)
-        assert action.status == 'completed'
-        if action.data['results']['Code'] != '0':
-            log('Command failed: ' + cmd)
-            log('unit: ' + unit.entity_id)
-            log('code: ' + action.data['results']['Code'])
-            log('stdout: ' + action.data['results']['Stdout'])
-            log('stderr: ' + action.data['results']['Stderr'])
-            raise Exception('Command failed: ' + cmd)
-        assert action.data['results']['Code'] == '0'
-        return action.data['results']['Stdout']
+    async def run_until_success(cmd):
+        while True:
```

Comment:
Forever is a big word shouldn't there be some long timeout

Reply:
Yeah, I was on the fence about this. I figured we could let the [global test timeout](https://github.com/juju-solutions/kubernetes-jenkins/blob/master/integration-tests/utils.py#L145) catch it, but I wouldn't be opposed to adding a timeout here too - no smaller than 10 min, though.